docs/cep/query-guide/custom-script-functions.md

```diff
@@ -0,0 +1,146 @@
+---
+sidebar_position: 80
+title: Custom Script Functions
+---
+
+Scripts allow you to write functions in JavaScript and execute them within queries for a stream. Functions defined with scripts can be accessed in queries similar to any other inbuilt function. You can use function definitions to define these scripts.
+
+Function parameters are passed into the function logic as an `Object[]` and with the name `data`.
+
+## Purpose
+
+Scripts allow you to define a function operation that is not provided in stream worker built-in functions. You do not need to write an extension to define the function logic.
+
+## Syntax
+
+The syntax for a script definition is as follows.
+
+```js
+define function <function name>[<javascript>] return <return type> {
+    <operation of the function>
+};
+```
+
+The defined function can be used in the queries similar to inbuilt functions as follows.
+
+```js
+<function name>( (<function parameter>(, <function parameter>)*)? )
+```
+
+The custom functions defined with these function definitions have higher precedence compared to other functions.
+
+## Parameters
+
+The following parameters are configured when defining a script.
+
+| Parameter     | Description |
+| ------------- |-------------|
+| function name | 	The name of the function. By convention, `camelCase` is used for the function name.|
+| language name | The name of the programming language used to define the script. Currently, Macrometa supports `javascript`.|
```

Comment:
```suggestion
| language name | The name of the programming language used to define the script. Macrometa supports `javascript`.|
```

Should we change this to say "`javascript` and `python`"?  Or do we only support JS in this particular instance?

Reply:
We only support JavaScript for this.